framework/src/application/node/transport/transport.ts

```diff
@@ -342,9 +343,17 @@ export class Transport {
 
 		if (idsNotInPool.length) {
 			// Check if any transaction that was not in the queues, is in the database instead.
-			const transactionsFromDatabase = await this._chainModule.dataAccess.getTransactionsByIDs(
-				idsNotInPool,
-			);
+			const transactionsFromDatabase: BaseTransaction[] = [];
+			for (const id of idsNotInPool) {
+				try {
+					const tx = await this._chainModule.dataAccess.getTransactionByID(id);
+					transactionsFromDatabase.push(tx);
+				} catch (error) {
+					if (!(error instanceof NotFoundError)) {
+						throw error;
+					}
+				}
+			}
```

Comment:
since the code repeats, how about a private function?

Reply:
I think https://github.com/LiskHQ/lisk-sdk/issues/5425 will cover this.